### ISSUE TITLE
plugin update goes wrong with vagrant 1.8.3 on mac osx

### ISSUE THREAD

**Issue by biemond** [mitchellh/vagrant#7415]
I get this error when I do an update of the plugins with version 1.8.3 on mac osx.

vagrant plugin update vagrant-vmware-fusion
or
vagrant plugin update vagrant-share 

```
Updating plugins: vagrant-share. This may take a few minutes...
/opt/vagrant/embedded/gems/gems/bundler-1.12.0/lib/bundler/resolver.rb:303:in `block in sort_dependencies': undefined method `payload' for nil:NilClass (NoMethodError)
	from /opt/vagrant/embedded/gems/gems/bundler-1.12.0/lib/bundler/resolver.rb:300:in `each'
	from /opt/vagrant/embedded/gems/gems/bundler-1.12.0/lib/bundler/resolver.rb:300:in `sort_by'
	from /opt/vagrant/embedded/gems/gems/bundler-1.12.0/lib/bundler/resolver.rb:300:in `sort_dependencies'
	from /opt/vagrant/embedded/gems/gems/bundler-1.12.0/lib/bundler/vendor/molinillo/lib/molinillo/resolution.rb:131:in `block (2 levels) in <class:Resolution>'
	from /opt/vagrant/embedded/gems/gems/bundler-1.12.0/lib/bundler/vendor/molinillo/lib/molinillo/resolution.rb:431:in `push_state_for_requirements'
	from /opt/vagrant/embedded/gems/gems/bundler-1.12.0/lib/bundler/vendor/molinillo/lib/molinillo/resolution.rb:423:in `require_nested_dependencies_for'
	from /opt/vagrant/embedded/gems/gems/bundler-1.12.0/lib/bundler/vendor/molinillo/lib/molinillo/resolution.rb:411:in `activate_spec'
	from /opt/vagrant/embedded/gems/gems/bundler-1.12.0/lib/bundler/vendor/molinillo/lib/molinillo/resolution.rb:353:in `attempt_to_swap_possibility'
	from /opt/vagrant/embedded/gems/gems/bundler-1.12.0/lib/bundler/vendor/molinillo/lib/molinillo/resolution.rb:333:in `attempt_to_activate_existing_spec'
	from /opt/vagrant/embedded/gems/gems/bundler-1.12.0/lib/bundler/vendor/molinillo/lib/molinillo/resolution.rb:318:in `attempt_to_activate'
	from /opt/vagrant/embedded/gems/gems/bundler-1.12.0/lib/bundler/vendor/molinillo/lib/molinillo/resolution.rb:147:in `process_topmost_state'
	from /opt/vagrant/embedded/gems/gems/bundler-1.12.0/lib/bundler/vendor/molinillo/lib/molinillo/resolution.rb:72:in `resolve'
	from /opt/vagrant/embedded/gems/gems/bundler-1.12.0/lib/bundler/vendor/molinillo/lib/molinillo/resolver.rb:42:in `resolve'
	from /opt/vagrant/embedded/gems/gems/bundler-1.12.0/lib/bundler/resolver.rb:201:in `start'
	from /opt/vagrant/embedded/gems/gems/bundler-1.12.0/lib/bundler/resolver.rb:184:in `resolve'
	from /opt/vagrant/embedded/gems/gems/bundler-1.12.0/lib/bundler/definition.rb:200:in `resolve'
	from /opt/vagrant/embedded/gems/gems/bundler-1.12.0/lib/bundler/definition.rb:140:in `specs'
	from /opt/vagrant/embedded/gems/gems/bundler-1.12.0/lib/bundler/definition.rb:129:in `resolve_remotely!'
	from /opt/vagrant/embedded/gems/gems/bundler-1.12.0/lib/bundler/installer.rb:195:in `resolve_if_need'
	from /opt/vagrant/embedded/gems/gems/bundler-1.12.0/lib/bundler/installer.rb:70:in `run'
	from /opt/vagrant/embedded/gems/gems/bundler-1.12.0/lib/bundler/installer.rb:22:in `install'
	from /opt/vagrant/embedded/gems/gems/vagrant-1.8.3/lib/vagrant/bundler.rb:230:in `block in internal_install'
	from /opt/vagrant/embedded/gems/gems/vagrant-1.8.3/lib/vagrant/bundler.rb:290:in `block in with_isolated_gem'
	from /opt/vagrant/embedded/lib/ruby/2.2.0/rubygems/user_interaction.rb:45:in `use_ui'
	from /opt/vagrant/embedded/gems/gems/vagrant-1.8.3/lib/vagrant/bundler.rb:289:in `with_isolated_gem'
	from /opt/vagrant/embedded/gems/gems/vagrant-1.8.3/lib/vagrant/bundler.rb:229:in `internal_install'
	from /opt/vagrant/embedded/gems/gems/vagrant-1.8.3/lib/vagrant/bundler.rb:149:in `update'
	from /opt/vagrant/embedded/gems/gems/vagrant-1.8.3/lib/vagrant/plugin/manager.rb:111:in `update_plugins'
	from /opt/vagrant/embedded/gems/gems/vagrant-1.8.3/plugins/commands/plugin/action/update_gems.rb:23:in `call'
	from /opt/vagrant/embedded/gems/gems/vagrant-1.8.3/lib/vagrant/action/warden.rb:34:in `call'
	from /opt/vagrant/embedded/gems/gems/vagrant-1.8.3/lib/vagrant/action/builder.rb:116:in `call'
	from /opt/vagrant/embedded/gems/gems/vagrant-1.8.3/lib/vagrant/action/runner.rb:66:in `block in run'
	from /opt/vagrant/embedded/gems/gems/vagrant-1.8.3/lib/vagrant/util/busy.rb:19:in `busy'
	from /opt/vagrant/embedded/gems/gems/vagrant-1.8.3/lib/vagrant/action/runner.rb:66:in `run'
	from /opt/vagrant/embedded/gems/gems/vagrant-1.8.3/plugins/commands/plugin/command/base.rb:14:in `action'
	from /opt/vagrant/embedded/gems/gems/vagrant-1.8.3/plugins/commands/plugin/command/update.rb:23:in `execute'
	from /opt/vagrant/embedded/gems/gems/vagrant-1.8.3/plugins/commands/plugin/command/root.rb:56:in `execute'
	from /opt/vagrant/embedded/gems/gems/vagrant-1.8.3/lib/vagrant/cli.rb:42:in `execute'
	from /opt/vagrant/embedded/gems/gems/vagrant-1.8.3/lib/vagrant/environment.rb:302:in `cli'
	from /opt/vagrant/embedded/gems/gems/vagrant-1.8.3/bin/vagrant:174:in `<main>'
```

**Comment by coka** [comment]
I get exactly the same error on OS X 10.11.5 with Ruby 2.0.0p648. I tried to install `vagrant-timezone`. This looks similar to #7073.

**Comment by sebastiaanluca** [comment]
Same for `vagrant plugin install vagrant-vbguest` (can't create a new VM as it can't mount the endpoints):

```
Installing the 'vagrant-vbguest' plugin. This can take a few minutes...
/opt/vagrant/embedded/gems/gems/bundler-1.12.0/lib/bundler/resolver.rb:303:in `block in sort_dependencies': undefined method `payload' for nil:NilClass (NoMethodError)
	from /opt/vagrant/embedded/gems/gems/bundler-1.12.0/lib/bundler/resolver.rb:300:in `each'
	from /opt/vagrant/embedded/gems/gems/bundler-1.12.0/lib/bundler/resolver.rb:300:in `sort_by'
	from /opt/vagrant/embedded/gems/gems/bundler-1.12.0/lib/bundler/resolver.rb:300:in `sort_dependencies'
	from /opt/vagrant/embedded/gems/gems/bundler-1.12.0/lib/bundler/vendor/molinillo/lib/molinillo/resolution.rb:131:in `block (2 levels) in <class:Resolution>'
	from /opt/vagrant/embedded/gems/gems/bundler-1.12.0/lib/bundler/vendor/molinillo/lib/molinillo/resolution.rb:431:in `push_state_for_requirements'
	from /opt/vagrant/embedded/gems/gems/bundler-1.12.0/lib/bundler/vendor/molinillo/lib/molinillo/resolution.rb:423:in `require_nested_dependencies_for'
	from /opt/vagrant/embedded/gems/gems/bundler-1.12.0/lib/bundler/vendor/molinillo/lib/molinillo/resolution.rb:411:in `activate_spec'
	from /opt/vagrant/embedded/gems/gems/bundler-1.12.0/lib/bundler/vendor/molinillo/lib/molinillo/resolution.rb:353:in `attempt_to_swap_possibility'
	from /opt/vagrant/embedded/gems/gems/bundler-1.12.0/lib/bundler/vendor/molinillo/lib/molinillo/resolution.rb:333:in `attempt_to_activate_existing_spec'
	from /opt/vagrant/embedded/gems/gems/bundler-1.12.0/lib/bundler/vendor/molinillo/lib/molinillo/resolution.rb:318:in `attempt_to_activate'
	from /opt/vagrant/embedded/gems/gems/bundler-1.12.0/lib/bundler/vendor/molinillo/lib/molinillo/resolution.rb:147:in `process_topmost_state'
	from /opt/vagrant/embedded/gems/gems/bundler-1.12.0/lib/bundler/vendor/molinillo/lib/molinillo/resolution.rb:72:in `resolve'
	from /opt/vagrant/embedded/gems/gems/bundler-1.12.0/lib/bundler/vendor/molinillo/lib/molinillo/resolver.rb:42:in `resolve'
	from /opt/vagrant/embedded/gems/gems/bundler-1.12.0/lib/bundler/resolver.rb:201:in `start'
	from /opt/vagrant/embedded/gems/gems/bundler-1.12.0/lib/bundler/resolver.rb:184:in `resolve'
	from /opt/vagrant/embedded/gems/gems/bundler-1.12.0/lib/bundler/definition.rb:200:in `resolve'
	from /opt/vagrant/embedded/gems/gems/bundler-1.12.0/lib/bundler/definition.rb:140:in `specs'
	from /opt/vagrant/embedded/gems/gems/bundler-1.12.0/lib/bundler/definition.rb:129:in `resolve_remotely!'
	from /opt/vagrant/embedded/gems/gems/bundler-1.12.0/lib/bundler/installer.rb:195:in `resolve_if_need'
	from /opt/vagrant/embedded/gems/gems/bundler-1.12.0/lib/bundler/installer.rb:70:in `run'
	from /opt/vagrant/embedded/gems/gems/bundler-1.12.0/lib/bundler/installer.rb:22:in `install'
	from /opt/vagrant/embedded/gems/gems/vagrant-1.8.3/lib/vagrant/bundler.rb:230:in `block in internal_install'
	from /opt/vagrant/embedded/gems/gems/vagrant-1.8.3/lib/vagrant/bundler.rb:290:in `block in with_isolated_gem'
	from /opt/vagrant/embedded/lib/ruby/2.2.0/rubygems/user_interaction.rb:45:in `use_ui'
	from /opt/vagrant/embedded/gems/gems/vagrant-1.8.3/lib/vagrant/bundler.rb:289:in `with_isolated_gem'
	from /opt/vagrant/embedded/gems/gems/vagrant-1.8.3/lib/vagrant/bundler.rb:229:in `internal_install'
	from /opt/vagrant/embedded/gems/gems/vagrant-1.8.3/lib/vagrant/bundler.rb:102:in `install'
	from /opt/vagrant/embedded/gems/gems/vagrant-1.8.3/lib/vagrant/plugin/manager.rb:62:in `block in install_plugin'
	from /opt/vagrant/embedded/gems/gems/vagrant-1.8.3/lib/vagrant/plugin/manager.rb:72:in `call'
	from /opt/vagrant/embedded/gems/gems/vagrant-1.8.3/lib/vagrant/plugin/manager.rb:72:in `install_plugin'
	from /opt/vagrant/embedded/gems/gems/vagrant-1.8.3/plugins/commands/plugin/action/install_gem.rb:37:in `call'
	from /opt/vagrant/embedded/gems/gems/vagrant-1.8.3/lib/vagrant/action/warden.rb:34:in `call'
	from /opt/vagrant/embedded/gems/gems/vagrant-1.8.3/lib/vagrant/action/builder.rb:116:in `call'
	from /opt/vagrant/embedded/gems/gems/vagrant-1.8.3/lib/vagrant/action/runner.rb:66:in `block in run'
	from /opt/vagrant/embedded/gems/gems/vagrant-1.8.3/lib/vagrant/util/busy.rb:19:in `busy'
	from /opt/vagrant/embedded/gems/gems/vagrant-1.8.3/lib/vagrant/action/runner.rb:66:in `run'
	from /opt/vagrant/embedded/gems/gems/vagrant-1.8.3/plugins/commands/plugin/command/base.rb:14:in `action'
	from /opt/vagrant/embedded/gems/gems/vagrant-1.8.3/plugins/commands/plugin/command/install.rb:32:in `block in execute'
	from /opt/vagrant/embedded/gems/gems/vagrant-1.8.3/plugins/commands/plugin/command/install.rb:31:in `each'
	from /opt/vagrant/embedded/gems/gems/vagrant-1.8.3/plugins/commands/plugin/command/install.rb:31:in `execute'
	from /opt/vagrant/embedded/gems/gems/vagrant-1.8.3/plugins/commands/plugin/command/root.rb:56:in `execute'
	from /opt/vagrant/embedded/gems/gems/vagrant-1.8.3/lib/vagrant/cli.rb:42:in `execute'
	from /opt/vagrant/embedded/gems/gems/vagrant-1.8.3/lib/vagrant/environment.rb:302:in `cli'
	from /opt/vagrant/embedded/gems/gems/vagrant-1.8.3/bin/vagrant:174:in `<main>'
```

**Comment by gildegoma** [comment]
Same problem when updating Vagrant plugins on Linux (Ubuntu)... but `vagrant up` command is also failing:

```shell
$ vagrant up
/tank/gildegoma/wip/support/vagrant/test-1.8.3/Vagrantfile:23:in `block (2 levels) in <top (required)>': undefined method ` ' for main:Object (NoMethodError)
	from /opt/vagrant/embedded/gems/gems/vagrant-1.8.3/plugins/kernel_v2/config/vm_provisioner.rb:72:in `call'
	from /opt/vagrant/embedded/gems/gems/vagrant-1.8.3/plugins/kernel_v2/config/vm_provisioner.rb:72:in `add_config'
	from /opt/vagrant/embedded/gems/gems/vagrant-1.8.3/plugins/kernel_v2/config/vm.rb:321:in `provision'
	from /tank/gildegoma/wip/support/vagrant/test-1.8.3/Vagrantfile:17:in `block in <top (required)>'
	from /opt/vagrant/embedded/gems/gems/vagrant-1.8.3/lib/vagrant/config/v2/loader.rb:37:in `call'
	from /opt/vagrant/embedded/gems/gems/vagrant-1.8.3/lib/vagrant/config/v2/loader.rb:37:in `load'
	from /opt/vagrant/embedded/gems/gems/vagrant-1.8.3/lib/vagrant/config/loader.rb:113:in `block (2 levels) in load'
	from /opt/vagrant/embedded/gems/gems/vagrant-1.8.3/lib/vagrant/config/loader.rb:107:in `each'
	from /opt/vagrant/embedded/gems/gems/vagrant-1.8.3/lib/vagrant/config/loader.rb:107:in `block in load'
	from /opt/vagrant/embedded/gems/gems/vagrant-1.8.3/lib/vagrant/config/loader.rb:104:in `each'
	from /opt/vagrant/embedded/gems/gems/vagrant-1.8.3/lib/vagrant/config/loader.rb:104:in `load'
	from /opt/vagrant/embedded/gems/gems/vagrant-1.8.3/lib/vagrant/vagrantfile.rb:28:in `initialize'
	from /opt/vagrant/embedded/gems/gems/vagrant-1.8.3/lib/vagrant/environment.rb:740:in `new'
	from /opt/vagrant/embedded/gems/gems/vagrant-1.8.3/lib/vagrant/environment.rb:740:in `vagrantfile'
	from /opt/vagrant/embedded/gems/gems/vagrant-1.8.3/lib/vagrant/environment.rb:486:in `host'
	from /opt/vagrant/embedded/gems/gems/vagrant-1.8.3/lib/vagrant/environment.rb:208:in `block in action_runner'
	from /opt/vagrant/embedded/gems/gems/vagrant-1.8.3/lib/vagrant/action/runner.rb:33:in `call'
	from /opt/vagrant/embedded/gems/gems/vagrant-1.8.3/lib/vagrant/action/runner.rb:33:in `run'
	from /opt/vagrant/embedded/gems/gems/vagrant-1.8.3/lib/vagrant/environment.rb:473:in `hook'
	from /opt/vagrant/embedded/gems/gems/vagrant-1.8.3/lib/vagrant/environment.rb:722:in `unload'
	from /opt/vagrant/embedded/gems/gems/vagrant-1.8.3/bin/vagrant:177:in `ensure in <main>'
	from /opt/vagrant/embedded/gems/gems/vagrant-1.8.3/bin/vagrant:177:in `<main>'
```

Not sure why the embedded Ruby is 2.2.0 ... I thought that Vagrant was [staying on 2.0](https://github.com/mitchellh/vagrant/blob/master/.travis.yml#L13)
`/opt/vagrant/embedded/lib/ruby/2.2.0/rubygems/user_interaction.rb`

**Comment by sethvargo** [comment]
Looks like this is related to the bundler API bump. 

This is a confirmed issue and regression. Please do not add additional +1 or me too. You can watch this issue for updates by clicking the "subscribe" link on the right. Thanks!

**Comment by mawatech** [comment]
This was the reason why I submitted PR 7404 :-(

**Comment by diglin** [comment]
Installing version 1.8.1 over 1.8.3 resolved the issue for me

**Comment by clari-patrick** [comment]
+1 for issue with vagrant-vbguest

**Comment by sethvargo** [comment]
@mawatech the upgrade to bundler is what _caused_ this - it was upgraded in a prior commit to your PR.

**Comment by akaSStalkALEX** [comment]
Sorry guys, that I'm asking, if I manually edit this file will it work?

**Comment by jvthert** [comment]
Same here. Nice the issue is fixed in the upcoming release but what about the current one?

**Comment by denisinla** [comment]
Do we need to downgrade Vagrant and bundler(1.12.5) for vagrant plugins to work again?

**Comment by sethvargo** [comment]
We have a release scheduled for this week to address this issue. You can use Vagrant 1.8.1 in the meantime, which does not have this issue.

**Comment by coka** [comment]
@sethvargo Is 1.8.4 the release you are referring to?

**Comment by m4olivei** [comment]
Upgraded to 1.8.4, and installing plugins totally works.  Thanks!

**Comment by mikeatlas** [comment]
Ditto, had this bug on 1.8.3 and it resolved with 1.8.4. https://www.vagrantup.com/downloads.html

**Comment by deshack** [comment]
I got the same issue while installing vagrant-share plugin (v1.1.5) on Ubuntu 16.04 with vagrant 1.8.1. Fixed by upgrading to vagrant 1.8.4 from yakkety repositories.
